pydra/engine/tests/test_submitter.py

```diff
@@ -575,6 +573,7 @@ def test_sge_no_limit_maxthreads(tmpdir):
     assert job_1_endtime > job_2_starttime
 
 
+@pytest.mark.xfail(reason="Not sure")
```

Comment:
Ah yes, this is that test I wrote to hit the exception handling code (which attempts to help find where the problem is). Not sure why it would no longer work, maybe we are copying the inputs somewhere

Reply:
Got it, just needed to implement the __bytes_repr__ to A. Not sure if it points to a case that we need to handle better with the hashing functions though.